### DISCUSSION TITLE
Don't create mutations for feature excluded code

### DISCUSSION THREAD

**Discussion by Azkellas** [sourcefrog/cargo-mutants#400] [Ideas]
I have a project that looks roughly like this:
```
src/
    - lib.rs
    - other_file.rs
```
where `libs.rs` contains
```rust
#[cfg(feature = "my_feature")]
mod other_file;

...
```

and `other_file` contains both logic and tests.

Curently `cargo mutants --no-default-features` generates mutants in `other_file.rs` that will not be compiled and not be tested. This means lost time testing nothing and investigating wrong misses.

I guess it's good practice the run `cargo mutants` with `--all-features` and it solves this issue but having feedback on mutations that were discarded (at compilation or ideally at generation) due to feature selection would be valuable imho.

You can find a concrete example in this crate: https://github.com/Azkellas/mesh_to_sdf/tree/main/mesh_to_sdf where `point/impl_*.rs` file are feature gated to avoid pulling dependencies.

**Comment by sourcefrog** [comment]
Thanks, yep, it does not really understand any `cfg` items at the moment, but I would like to add it. There is https://github.com/sourcefrog/cargo-mutants/issues/50 about this.  Maybe it's useful to consider features as an important special case, distinct from cfg on OS or platform attributes.

> having feedback on mutations that were discarded (at compilation or ideally at generation) due to feature selection would be valuable imho.

Actually this is interesting and a bit different from just knowing not to traverse the mod statement if the feature is off. I'll have to think about that.
